### PR TITLE
Supplemental Claims | Fix a11y issues on review & submit

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

- *(Summarize the changes that have been made to the platform)*
  > - Wrap contestable issue & issue summary page content in a `fieldset`/`legend`
  > - Fix header levels in contestable issues edit mode
  > - Hide issue summary and evidence pages that aren't the evidence summary (matches design)
  > - Fix private evidence limitation path (see [#50125 (comment)](https://github.com/department-of-veterans-affairs/va.gov-team/issues/50125#issuecomment-1363997607))
  > - Fix issue & evidence summary focus management
  > - Update unit & e2e tests
  > - Various design fixes to margin & padding
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution)*
- *(Which team do you work for, does your team own the maintainance of this component?)*
  > Benefits team 1
- *(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)*

## Related issue(s)

[#48403](https://github.com/department-of-veterans-affairs/va.gov-team/issues/48403)

## Testing done

- *Describe what the old behavior was prior to the change*
  > - Header level skipping while in edit mode on the review & submit page
  > - Focus lost after edit & after updating
  > - Design mismatch
- *Describe the steps required to verify your changes are working as expected*
  > - Manual testing with HeadingsMap extension
  > - Manual focus management testing
- *Describe the tests completed and the results*
  > - Unit test updated and all passing

### After

<details><summary>New and relevant evidence accordion with supporting evidence in view mode</summary>

New and relevant evidence accordion expanded with notice of acknowledgement page and supporting evidence pages in view mode. HeadingsMap extension is showing the header levels: h3 for the accordion title, h4 for both page titles and 3 h5's nested under the supporting evidence</details>

<details><summary>New and relevant evidence accordion with supporting evidence in edit mode</summary>

Note that the original "Supporting evidence" h4 is hidden along with the edit button. The h4 shown in this screenshot is added on the review & submit page only in edit mode

New and relevant evidence accordion expanded with notice of acknowledgement page in view mode. The supporting evidence page is in edit mode and shows a newly added h4 Supporting evidence header - it is usually hidden in edit mode - with the 3 nested h5's</details>

## What areas of the site does it impact?

Supplemental Claim form (not yet in production)

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/appeals/995/tests/components/EvidenceVaRecords.unit.spec.jsx [hunk 1]
```diff
@@ -79,7 +72,7 @@ describe('<EvidenceVaRecords>', () => {
     });
 
     // close modal by clicking method-assigned hidden button
-    fireEvent.click($('#test-method', container), mouseClick);
```
Comment (jerekshoe) [inline src/applications/appeals/995/tests/components/EvidenceVaRecords.unit.spec.jsx:82]: What prompted this change? Was the mouseClick just not needed?

Comment (Mottie) [reply src/applications/appeals/995/tests/components/EvidenceVaRecords.unit.spec.jsx:82]: `fireEvent.click` already uses a `mouseClick`, so it's duplicated effort. If you need a custom event, then use `fireEvent(..., new CustomEvent(...))`

Comment (jerekshoe) [reply src/applications/appeals/995/tests/components/EvidenceVaRecords.unit.spec.jsx:82]: Ok cool, I assumed it was something like that, but was just curious. Thanks for the explanation!

Comment (Mottie) [reply src/applications/appeals/995/tests/components/EvidenceVaRecords.unit.spec.jsx:82]: Actually, [I think the second parameter for named events](https://testing-library.com/docs/guide-events/) is ignored anyway...

**Review by jerekshoe — APPROVED** [review]

**Pull request merged**